zulipterminal/helper.py

```diff
@@ -84,14 +84,13 @@ def wrapper(*args: Any, **kwargs: Any) -> Any:
     return wrapper
 
 
-def set_count(id_list: List[int], controller: Any, new_count: int) -> None:
-    # This method applies new_count for 'new message' (1) or 'read' (-1)
-    # (we could ensure this in a different way by a different type)
-    assert new_count == 1 or new_count == -1
-
-    messages = controller.model.index['messages']
-    unread_counts = controller.model.unread_counts  # type: UnreadCounts
-
+def _set_count_in_model(id_list: List[int], controller: Any, new_count: int,
+                        messages: Dict[int, Message],
+                        unread_counts: UnreadCounts) -> None:
+    """
+        This function contains the logic for setting the count in the model.
+    """
+    print(messages, flush=True)
     for id in id_list:
         msg = messages[id]
```

Comment:
Seeing this now, all we need is the messages that correspond to `id_list`, so we could simplify these function parameters further, potentially?

The same applies to the other function too.